io/plugins/eu.esdihumboldt.hale.io.json/src/eu/esdihumboldt/hale/io/json/TopoJSONInstanceWriter.java

```diff
@@ -0,0 +1,129 @@
+ * available under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution. If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * Contributors:
+ *     wetransform GmbH <http://www.wetransform.to>
+ */
+
+package eu.esdihumboldt.hale.io.json;
+
+import java.io.IOException;
+import java.net.URI;
+
+import eu.esdihumboldt.hale.common.core.io.IOProviderConfigurationException;
+import eu.esdihumboldt.hale.common.core.io.ProgressIndicator;
+import eu.esdihumboldt.hale.common.core.io.report.IOReport;
+import eu.esdihumboldt.hale.common.core.io.report.IOReporter;
+import eu.esdihumboldt.hale.common.core.io.report.impl.IOMessageImpl;
+import eu.esdihumboldt.hale.common.instance.geometry.impl.CodeDefinition;
+import eu.esdihumboldt.hale.common.instance.model.InstanceCollection;
+import eu.esdihumboldt.hale.common.schema.geometry.CRSDefinition;
+import eu.esdihumboldt.hale.io.shp.writer.ShapefileInstanceWriter;
+import json.topojson.api.TopojsonApi;
+
+/**
+ * Writes instances as TopoJSON.
+ * 
+ * @author Flaminia Catalli
+ */
+public class TopoJSONInstanceWriter extends JsonInstanceWriter {
+
+	private final CRSDefinition targetCrs = new CodeDefinition("EPSG:4326", true); // TODO:
+																					// check
+																					// if
+																					// CRS
+																					// is
+																					// correct
+
+//	/**
+//	 * Parameter name for the default geometry association.
+//	 * 
+//	 * @deprecated as of release 4.2.0 because we don't use geometry
+//	 *             configuration for geoJson and export the data in WG84 format.
+//	 *             As geoJson expects WGS 84 with lon/lat (see
+//	 *             https://tools.ietf.org/html/rfc7946)
+//	 */
+//	@Deprecated
+//	public static final String PARAM_GEOMETRY_CONFIG = "geojson.geometry.config";
+
+	/**
+	 * By default do not use geoJson or topoJson features for the output.
+	 */
+	public TopoJSONInstanceWriter() {
+		super(false, true);
+	}
+
+	@Override
+	protected String getDefaultTypeName() {
+		return "TopoJSON";
+
+	}
+
+	/**
+	 * @see eu.esdihumboldt.hale.io.json.JsonInstanceWriter#execute(eu.esdihumboldt.hale.common.core.io.ProgressIndicator,
+	 *      eu.esdihumboldt.hale.common.core.io.report.IOReporter)
+	 */
+	@Override
+	protected IOReport execute(ProgressIndicator progress, IOReporter reporter)
+			throws IOProviderConfigurationException, IOException {
+
+		progress.begin("Generating " + getDefaultTypeName(), ProgressIndicator.UNKNOWN);
+
+		try {
+			// 1. write instances into a shape file
+			// 2. convert the shape file into topoJson
+			URI location = getTarget().getLocation();
+			InstanceCollection instances = getInstances();
+			writeInstanceCollectionToShp(instances, progress, reporter, location);
```

Comment:
`location` is the final target of the TopoJSON output, it can't be used as the target for `ShapefileInstanceWriter`. As writing the Shapefile is an intermediary step, it should be written to a temporary file instead.

You could e.g. use `File.createTempFile(...)` to create a temporary file for the Shapefile writer.